assets/test/test_ini_file.js

```diff
@@ -0,0 +1,177 @@
+                                if (!err) {
+                                    assert(recurseBundleIni([dirPath1, dirPath2], [], "/").length === 2);
+                                    done();
+                                }
+                            });
+                        });
+                    }
+                });
+            });
+        });
+    });
+    describe('one valid input file, output', function() {
+        it('should have one entry', function (done) {
+            temp.mkdir("dir1", function(err, dirPath1){
+                fs.writeFile(path.join(dirPath1, "bundles.ini"), "[test_plugin]", function (err) {
+                    if (!err) {
+                        temp.mkdir("dir2", function(err, dirPath2){
+                            fs.writeFile(path.join(dirPath2, "bundles.ini"), "[test_plugin]\nentry_file = this.js\n", function (err) {
+                                if (!err) {
+                                    assert(recurseBundleIni([dirPath1, dirPath2], [], "/").length === 1);
+                                    done();
+                                }
+                            });
+                        });
+                    }
+                });
+            });
+        });
+    });
+    describe('no valid input files, output', function() {
+        it('should no entries', function (done) {
+            temp.mkdir("dir1", function(err, dirPath1){
+                fs.writeFile(path.join(dirPath1, "bundles.ini"), "[test_plugin]", function (err) {
+                    if (!err) {
+                        temp.mkdir("dir2", function(err, dirPath2){
+                            fs.writeFile(path.join(dirPath2, "bundles.ini"), "[test_plugin]", function (err) {
+                                if (!err) {
+                                    assert(recurseBundleIni([dirPath1, dirPath2], [], "/").length === 0);
+                                    done();
+                                }
+                            });
+                        });
+                    }
+                });
+            });
+        });
+    });
+    describe('nested folders with one ini file, output', function() {
+        it('should have one entry', function (done) {
+            temp.mkdir("dir1", function(err, dirPath1){
+                fs.mkdir(path.join(dirPath1, "sub"), function (err) {
+                    if (!err) {
+                        fs.writeFile(path.join(dirPath1, "sub", "bundles.ini"), "[test_plugin]\nentry_file = this.js\n", function (err) {
+                            if (!err) {
+                                assert(recurseBundleIni([dirPath1], [], "/").length === 1);
+                                done();
+                            }
+                        });
+                    }
+                });
+            });
+        });
+    });
+    describe('two valid input files with an external, output', function() {
```

Comment:
Will most of our unit tests be this nested, or is this just special to the ini parsing functions?